### ISSUE TITLE
Usage returning "Internal Server Error".  (5m billing)

### ISSUE THREAD

**Issue by rob0101** [amberelectric/public-api#232]
My script to fetch usage data is returning an "Internal Server Error" today.

https://api.amber.com.au/v1/sites/XXXXX/usage?startDate=2024-11-10&endDate=2024-12-09' -H 'accept: application/json' -H 'Authorization: Bearer psk_XXXXX'

I'm on the 5m beta, but this api call has been working ok with mixed 30m and 5m data.

**Comment by madpilot** [comment]
You're hitting an internal payload limit...

Short term: don't pull a full month's worth of data.

Longer term: We'll setup paging.

**Comment by rob0101** [comment]
Ah, so it's the volume of data this time.   More datapoints in the same timeframe has now exceeded some byte count.

I'll reduce the number of days.

Thanks.

**Comment by madpilot** [comment]
We're going to clamp this and the price API to 7-days at a time.

**Comment by madpilot** [comment]
> Ah, so it's the volume of data this time. More datapoints in the same timeframe has now exceeded some byte count.
> 
> I'll reduce the number of days.
> 
> Thanks.

Yeah, as the days since your 5-minute cutover is getting bigger, the amount of data required to fulfil the request has gotten bigger.

Why are you fetching such large data ranges continuously?

**Comment by rob0101** [comment]
Keep in mind that some may want to fetch this data at irregular times and need to do a bulk catchup.

I sometimes go more than a week between calls, and want to catch every day I've missed.

**Comment by rob0101** [comment]
I also like to fetch older data because that data might get corrected, e.g. initially missing and populated with filler data.

**Comment by madpilot** [comment]
> Keep in mind that some may want to fetch this data at irregular times and need to do a bulk catchup.
> 
> I sometimes go more than a week between calls, and want to catch every day I've missed.

You can do that by adjusting the startDate and endDate - it just means you'll need to make multiple queries to get the same amount of data.

**Comment by rob0101** [comment]
Yup.    1 month used to work, so that's what I went with to catch as many data updates as possible.
